src/vs/workbench/browser/parts/views/treeView.ts

```diff
@@ -1187,6 +1206,21 @@ class TreeRenderer extends Disposable implements ITreeRenderer<ITreeItem, FuzzyS
 		disposableStore.add(toDisposable(() => this.treeViewsService.removeRenderedTreeItemElement(node)));
 	}
 
+	private renderCheckbox(node: ITreeItem, templateData: ITreeExplorerTemplateData, disposableStore: DisposableStore) {
+		if (node.checkboxChecked !== undefined) {
+			if (!templateData.checkbox) {
+				const checkbox = new TreeItemCheckbox(templateData.checkboxContainer, this.checkboxStateHandler, this.themeService);
+				templateData.checkbox = checkbox;
+				disposableStore.add({ dispose: () => { checkbox.dispose(); templateData.checkbox = undefined; } });
```

Comment:
There's a `toDisposable` utility that you can use: https://github.com/microsoft/vscode/blob/9ef2ee5ce1abc7812bea03375b47af44ec7d46af/extensions/git/src/util.ts#L28-L30